addons/account/models/account_code_mapping.py

```diff
@@ -36,7 +36,7 @@ def _search(self, domain, offset=0, limit=None, order=None) -> Query:
             case [('account_id', 'in', account_ids)]:
                 self_ids = {
                     f'{account_id},{company.id}'
-                    for company in self.env.companies
+                    for company in self.env.user.company_ids
```

Comment:
This feels like the opposite of `display_mapping_tab`.
Why are we displaying all the companies if we only display them when we have at least two companies selected?